content/en/docs/guides/security/credential_reloading_file_watcher.md

```diff
@@ -0,0 +1,194 @@
+
+Please note that **the APIs are experimental so the usages in this document are 
+subject to be changed**. While we welcome more users to try and use the new API, 
+it should also be kept in mind that we don't guarantee the backward 
+compatibility of these APIs.
+
+
+### C++
+
+Credential reloading is configured through TLS options. It can be used on both 
+client and server side.
+
+On the client side,
+
+``` cpp
+grpc::experimental::TlsChannelCredentialsOptions options;
+
+// set up credential reloading in options
+auto certificate_provider = std::make_shared<FileWatcherCertificateProvider>(
+      CLIENT_KEY_PATH, CLIENT_CERT_PATH, CA_CERT_PATH, /*refresh_interval_sec=*/1);
+grpc::experimental::TlsChannelCredentialsOptions options;
+options.set_certificate_provider(certificate_provider);
+// Tells gRPC to watch the updates of the credentials
+options.watch_root_certs();
+options.watch_identity_key_cert_pairs();
+// kRootCertName and kIdentityCertName don't matter in this case - it can be any string that you wish to have as a alias for the cert name, or just use an empty string
+options.set_root_cert_name(kRootCertName);
+options.set_identity_cert_name(kIdentityCertName);
+auto channel_credentials = grpc::experimental::TlsCredentials(options);
+
+// Use options to create the credential
+auto channel_creds = grpc::experimental::TlsCredentials(options);
+auto channel = grpc::CreateChannel("myservice.example.com", channel_creds);
+std::unique_ptr<Greeter::Stub> stub(Greeter::NewStub(channel));
+// ...
+```
+On the server side,
+
+``` cpp
+// provider is mandatory for the server - see next section for details
+auto certificate_provider = std::make_shared<FileWatcherCertificateProvider>(...);
+grpc::experimental::TlsServerCredentialsOptions options(certificate_provider);
+
+// set up credential reloading in options
+auto certificate_provider = std::make_shared<FileWatcherCertificateProvider>(
+      SERVER_KEY_PATH, SERVER_CERT_PATH, CA_CERT_PATH, /*refresh_interval_sec=*/1);
+grpc::experimental::TlsChannelCredentialsOptions options;
+options.set_certificate_provider(certificate_provider);
+// Tells gRPC to watch the updates of the credentials
+options.watch_root_certs();
+options.watch_identity_key_cert_pairs();
+// kRootCertName and kIdentityCertName don't matter in this case - it can be any string that you wish to have as a alias for the cert name, or just use an empty string
+options.set_root_cert_name(kRootCertName);
+options.set_identity_cert_name(kIdentityCertName);
+auto channel_credentials = grpc::experimental::TlsCredentials(options);
+
+// Use options to create the credential
+auto server_credentials = grpc::experimental::TlsServerCredentials(options);
+GreeterServiceImpl service;
+ServerBuilder builder;
+builder.AddListeningPort("0.0.0.0:50051", server_credentials);
+builder.RegisterService(&service);
+std::unique_ptr<Server> server(builder.BuildAndStart());
+// ...
+```
+
+### Java
```

Comment:
The Java examples don't do credential reloading. They are currently one-shot.

Reply:
Ah sorry, those were the wrong snippets. Updated!